src/status_im/profile/core.cljs

```diff
@@ -85,11 +86,23 @@
   [_ value]
   {:profile/share-profile-link value})
 
+
+(rf/defn contact-built
+  {:events [:contacts/contact-built]}
+  [{:keys [db]} identity contact]
+  {:db       (assoc-in db [:contacts/contacts identity] contact)
+   :dispatch [:open-modal :profile]})
+
 (rf/defn show-profile
   {:events [:chat.ui/show-profile]}
   [{:keys [db]} identity ens-name]
   (let [my-public-key (get-in db [:multiaccount :public-key])]
     (when (not= my-public-key identity)
-      {:db (-> db
-               (assoc :contacts/identity identity)
-               (assoc :contacts/ens-name ens-name))})))
+      {:db            (-> db
+                          (assoc :contacts/identity identity)
+                          (assoc :contacts/ens-name ens-name))
+       :json-rpc/call [{:method     "wakuext_buildContact"
+                        :params     [identity]
+                        :on-success #(rf/dispatch [:contacts/contact-built
+                                                   identity
+                                                   (data-store.contacts/<-rpc %)])}]})))
```

Comment:
If I were doing this, I'd use the name `show-profile-success` instead of `contact-built` because from the caller might not be as interested in knowing the implementation details (that a contact was built), just that the call was successful.